modules/common/src/main/java/org/opencastproject/job/api/AbstractJobProducer.java

```diff
@@ -173,22 +173,32 @@ public boolean isReadyToAccept(Job job) throws ServiceRegistryException, Undispa
     // Note: We are not adding the job load in the next line because it is already accounted for in the load values we
     // get back from the service registry.
     float currentLoad = getServiceRegistry().getOwnLoad();
+    logger.debug("{} Current load on this host: {}, job's load: {}, job's status: {}, max load: {}", new Object[] {
+            Thread.currentThread().getId(), currentLoad, job.getJobLoad(), job.getStatus().name(),
+            maxload.getLoadFactor() });
+    // Add the current job load to compare below
+    currentLoad += job.getJobLoad();
 
     /* Note that this first clause looks at the *job's*, the other two look at the *node's* load
      * We're assuming that if this case is true, then we're also the most powerful node in the system for this service,
      * per the current job dispatching code in ServiceRegistryJpaImpl */
     if (job.getJobLoad() > maxload.getLoadFactor() && acceptJobLoadsExeedingMaxLoad) {
-      logger.warn("Accepting job {} of type {} with load {} even though load of {} is above this node's limit of {}.",
-              new Object[] { job.getId(), job.getJobType(), df.format(job.getJobLoad()), df.format(currentLoad), df.format(maxload.getLoadFactor()) });
+      logger.warn(
+              "{} Accepting job {} of type {} with load {} even though load of {} is above this node's limit of {}.",
+              new Object[] { Thread.currentThread().getId(), job.getId(), job.getJobType(), df.format(job.getJobLoad()),
+                      df.format(currentLoad), df.format(maxload.getLoadFactor()) });
       logger.warn("This is a configuration issue that you should resolve in a production system!");
       return true;
     } else if (currentLoad > maxload.getLoadFactor()) {
-      logger.debug("Declining job {} of type {} with load {} because load of {} would exceed this node's limit of {}.",
-              new Object[] { job.getId(), job.getJobType(), df.format(job.getJobLoad()), df.format(currentLoad), df.format(maxload.getLoadFactor()) });
+      logger.debug(
+              "{} Declining job {} of type {} with load {} because load of {} would exceed this node's limit of {}.",
+              new Object[] { Thread.currentThread().getId(), job.getId(), job.getJobType(), df.format(job.getJobLoad()),
```

Comment:
Don't use an array here. Just remove the `new Object[] { }` and it will still work.